---
layout: page
title: "Walkthroughs"
group: navigation
---

{% include JB/setup %}

### Officially supported walkthroughs

The following walkthroughs were created by the __sleuth__ team and demonstrate several use cases of __sleuth__. The code for generating each walkthrough is available on [GitHub](https://github.com/pachterlab/sleuth_walkthroughs).

- [Getting started with __sleuth__](https://pachterlab.github.io/sleuth_walkthroughs/trapnell/analysis.html)
  - Install and set up __sleuth__
  - How to perform a two condition differential analysis
  - Linking transcript IDs to genes
  - __sleuth__ live
  - Basic QC
- [Accounting for batch effects](https://pachterlab.github.io/sleuth_walkthroughs/bottomly/analysis.html)
  - Accounting for known batch effects
  - Extracting effect sizes (fold changes)
- [Differential analysis with multiple conditions and covariates](https://pachterlab.github.io/sleuth_walkthroughs/boj/analysis.html)
  - Testing frameworks for experimental designs with multiple conditions and covariates
  - Performing gene-level differnetial analysis
  - Identifying and removing outliers

### Community contributed walkthroughs

- [Combining data from multiple RNA-Seq experiments](http://achri.blogspot.com/2017/03/combining-rnaseq-experiments-to-find.html)
- [Timecourse analysis](http://www.nxn.se/valent/timecourse-analysis-with-sleuth)
