
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>Manual</title>
    <meta name="description" content="">
    <meta name="author" content="Pachter Lab">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Le styles -->
    <link href="/assets/themes/twitter/bootstrap/css/bootstrap.2.2.2.min.css" rel="stylesheet">
    <link href="/assets/themes/twitter/css/style.css?body=1" rel="stylesheet" type="text/css" media="all">

    <!-- Le fav and touch icons -->
  <!-- Update these with your own images
    <link rel="shortcut icon" href="images/favicon.ico">
    <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
  -->

    <!-- atom & rss feed -->
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">

  </head>

  <body>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container-narrow">
          <a class="brand" href="/">sleuth</a>
          <ul class="nav">
            
            
            


  
    
      
      	
      	<li><a href="/about.html">About</a></li>
      	
      
    
  
    
      
    
  
    
      
      	
      	<li><a href="/download.html">Download</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="/faq.html">FAQ</a></li>
      	
      
    
  
    
  
    
      
      	
      	<li><a href="/license.html">License</a></li>
      	
      
    
  
    
      
      	
      	<li class="active"><a href="/manual.html" class="active">Manual</a></li>
      	
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	<li><a href="/source.html">Source</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="/starting.html">Getting Started</a></li>
      	
      
    
  



          </ul>
        </div>
      </div>
    </div>

    <div class="container-narrow">

      <div class="content">
        
<div class="page-header">
  <h1>Manual </h1>
</div>

<div class="row-fluid">
  <div class="span12">
    
<p>Typing <code>kallisto</code> produces a list of usage options, which are:</p>

<pre><code>kallisto 0.42.2

Usage: kallisto &lt;CMD&gt; [arguments] ..

Where &lt;CMD&gt; can be one of:

    index         Builds a kallisto index
    quant         Runs the quantification algorithm
    h5dump        Converts HDF5-formatted results to plaintext
    version       Prints version information

Running kallisto &lt;CMD&gt; without arguments prints usage information for &lt;CMD&gt;

</code></pre>
<p>The four usage commands are:</p>

<h4 id="index">index</h4>

<p><code>kallisto index</code> builds an index from a FASTA formatted file of target sequences. The arguments for the index command are:</p>

<pre><code>kallisto 0.42.2
Builds a kallisto index

Usage: kallisto index [arguments] FASTA-files

Required argument:
-i, --index=STRING          Filename for the kallisto index to be constructed

Optional argument:
-k, --kmer-size=INT         k-mer (odd) length (default: 31, max value: 31)
</code></pre>

<p>The Fasta file supplied can be either in plaintext or gzipped format.</p>

<h4 id="quant">quant</h4>

<p><code>kallisto quant</code> runs the quantification algorithm. The arguments for the quant command are:</p>

<pre><code>kallisto 0.42.2
Computes equivalence classes for reads and quantifies abundances

Usage: kallisto quant [arguments] FASTQ-files

Required arguments:
-i, --index=STRING            Filename for the kallisto index to be used for
                              quantification
-o, --output-dir=STRING       Directory to write output to

Optional arguments:
    --single                  Quantify single-end reads
    --bias                    Perform sequence based bias correction
-l, --fragment-length=DOUBLE  Estimated average fragment length
                              (default: value is estimated from the input data)
    --pseudobam               Output pseudoalignments in SAM format to stdout
-b, --bootstrap-samples=INT   Number of bootstrap samples (default: 0)
-t, --threads=INT             Number of threads to use for bootstraping (default: 1)
    --seed=INT                Seed for the bootstrap sampling (default: 42)
    --plaintext               Output plaintext instead of HDF5
</code></pre>

<p><strong>kallisto</strong> can process either single-end or paired-end reads. The default running mode is paired-end and requires an even number of FASTQ files represented as pairs, e.g.</p>

<pre><code>kallisto quant -i index -o output pairA_1.fastq pairA_2.fastq pairB_1.fastq pairB_2.fastq
</code></pre>

<p>For single-end mode you supply the <code>--single</code> flag and list any number of FASTQ files, e.g</p>

<pre><code>kallisto quant -i index -o output --single -l 180 file1.fastq.gz file2.fastq.gz file3.fastq.gz
</code></pre>

<p>FASTQ files can be either plaintext or
gzipped.</p>

<p><strong>Important note:</strong> only supply one sample at a time to <em>kallisto</em>. The
multiple FASTQ (pair) option is for users who have samples that span multiple
FASTQ files.</p>

<p>In the case of single-end reads, the -l option must be used to specify the average fragment length. Typical Illumina libraries produce fragment lengths ranging from 180–200 bp but it’s best to determine this from a library quantification with an instrument such as an Agilent Bioanalyzer. For paired-end reads, the average fragment length can be directly estimated from the reads and the program will do so if -l is not used (this is the preferred run mode).</p>

<p>The number of bootstrap samples is specified using -b. Note that because of the large amount of data that may be produced when the number of bootstrap samples is high, <strong>kallisto</strong> outputs bootstrap results in HDF5 format. The <code>h5dump</code> command can be used afterwards to convert this output to plaintext, however most convenient is to analyze bootstrap results with <strong>sleuth</strong>.</p>

<p><code>kallisto quant</code> produces three <strong>output</strong> files by default:</p>

<ul>
  <li><strong>abundances.h5</strong> is a HDF5 binary file containing run info, abundance
esimates, bootstrap estimates, and transcript length information length. This
file can be read in by <strong>sleuth</strong></li>
  <li><strong>abundances.tsv</strong> is a plaintext file of the abundance estimates. It does
not contains bootstrap estimates. Please use the <code>--plaintext</code> mode to output
plaintext abundance estimates. Alternatively, <code>kallisto h5dump</code> can be used
to output an HDF5 file to plaintext. The first line contains a header for
each column, including <a href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/">estimated counts, TPM, effective length</a>.</li>
  <li><strong>run_info.json</strong> is a json file containing information about the run</li>
</ul>

<h5 id="optional-arguments">Optional arguments</h5>

<ul>
  <li>
    <p><code>--bias</code> learns parameters for a model of sequences specific bias and corrects the abundances accordlingly.</p>
  </li>
  <li>
    <p><code>-t, --threads</code> specifies the number of threads to be used while running bootstrap. The default value is 1 thread, specifying more than the number of bootstraps or the number of cores on your machine has no additional effect.</p>
  </li>
</ul>

<h5 id="pseudobam">Pseudobam</h5>

<p><code>--pseudobam</code> outputs all pseudoalignments in SAM format to the standard output. The stream can either be redirected into a file, or converted to bam using <code>samtools</code>.</p>

<p>For example</p>

<pre><code>kallisto quant -i index -o out --pseudobam r1.fastq r2.fastq &gt; out.sam
</code></pre>

<p>or by piping directly into <code>samtools</code></p>

<pre><code>kallisto quant -i index -o out --pseudobam r1.fastq r2.fastq | samtools view -Sb - &gt; out.bam
</code></pre>

<p>A detailed description of the SAM output is <a href="pseudobam.html">here</a>.</p>

<h4 id="h5dump">h5dump</h4>

<p><code>kallisto h5dump</code> converts
<a href="https://www.hdfgroup.org/HDF5/whatishdf5.html">HDF5</a>-formatted results to
plaintext. The arguments for the h5dump command are:</p>

<pre><code>kallisto 0.42.2
Converts HDF5-formatted results to plaintext

Usage:  kallisto h5dump [arguments] abundance.h5

Required argument:
-o, --output-dir=STRING       Directory to write output to

</code></pre>

<h4 id="version">version</h4>

<p><code>kallisto version</code> displays the current version of the software.</p>

  </div>
</div>


      </div>
      <hr>
      <footer>
        <p>&copy; 2015 Pachter Lab
          with help from <a href="http://jekyllbootstrap.com" target="_blank" title="The Definitive Jekyll Blogging Framework">Jekyll Bootstrap</a>
          and <a href="http://twitter.github.com/bootstrap/" target="_blank">Twitter Bootstrap</a>
        </p>
      </footer>

    </div>

    


  </body>
</html>

