
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>Getting Started</title>
    <meta name="description" content="">
    <meta name="author" content="Pachter Lab">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Le styles -->
    <link href="/assets/themes/twitter/bootstrap/css/bootstrap.2.2.2.min.css" rel="stylesheet">
    <link href="/assets/themes/twitter/css/style.css?body=1" rel="stylesheet" type="text/css" media="all">

    <!-- Le fav and touch icons -->
  <!-- Update these with your own images
    <link rel="shortcut icon" href="images/favicon.ico">
    <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
  -->

    <!-- atom & rss feed -->
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">

  </head>

  <body>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container-narrow">
          <a class="brand" href="/">sleuth</a>
          <ul class="nav">
            
            
            


  
    
      
      	
      	<li><a href="/about.html">About</a></li>
      	
      
    
  
    
      
    
  
    
      
      	
      	<li><a href="/download.html">Download</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="/faq.html">FAQ</a></li>
      	
      
    
  
    
  
    
      
      	
      	<li><a href="/license.html">License</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="/manual.html">Manual</a></li>
      	
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	<li><a href="/source.html">Source</a></li>
      	
      
    
  
    
      
      	
      	<li class="active"><a href="/starting.html" class="active">Getting Started</a></li>
      	
      
    
  



          </ul>
        </div>
      </div>
    </div>

    <div class="container-narrow">

      <div class="content">
        
<div class="page-header">
  <h1>Getting Started </h1>
</div>

<div class="row-fluid">
  <div class="span12">
    
<p>The short tutorial below explains how to run <strong>kallisto</strong> using a small example distributed with the program. If you already know how to use <strong>kallisto</strong> but just want to get it quickly installed on your Mac, the easiest way to do so is using brew. Type:</p>

<p><code>ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"</code> to install brew, followed by</p>

<p><code>brew tap homebrew/science</code> and then</p>

<p><code>brew install kallisto</code></p>

<p>to have kallisto universally executable.</p>

<h4 id="download">Download</h4>

<p>The example/test distributed with <strong>kallisto</strong> is included with the binaries we distribute with the program:</p>

<ul>
  <li>
    <p>For Mac download the <strong>kallisto</strong> binary from <a href="https://github.com/pachterlab/kallisto/releases/download/v0.42.1/kallisto_mac-v0.42.1.tar.gz">here</a></p>
  </li>
  <li>
    <p>For Ubuntu download the <strong>kallisto</strong> binary from <a href="https://github.com/pachterlab/kallisto/releases/download/v0.42.1/kallisto_ubuntu-v0.42.1.tar.gz">here</a></p>
  </li>
</ul>

<h4 id="install">Install</h4>

<p>If you do not already have <strong>kallisto</strong> universally executable on your machine, Begin by copying the <strong>kallisto</strong> executable from the downloaded binary to</p>

<p><code>cp kallisto /usr/local/bin/</code></p>

<p>Depending on your system configuration, you might need to prepend the command
with <code>sudo</code>.</p>

<p>You should be able to type <code>kallisto</code> and see:</p>

<pre><code>kallisto 0.42.1

Usage: kallisto &lt;CMD&gt; [arguments] ..

Where &lt;CMD&gt; can be one of:

    index         Builds a kallisto index
    quant         Runs the quantification algorithm
    h5dump        Converts HDF5-formatted results to plaintext
    version       Prints version information

Running kallisto &lt;CMD&gt; without arguments prints usage information for &lt;CMD&gt;
</code></pre>

<h4 id="building-an-index">Building an index</h4>

<p><strong>kallisto</strong> quantifies read files directly without the need for read alignment, but it does perform a procedure called pseudoalignment. Pseudoalignment requires processing a transcriptome file to create a “transcriptome index”. To begin, first change directories to where the test files distributed with the <strong>kallisto</strong> executable are located:</p>

<p><code>cd kallisto/tests</code></p>

<p>Next, build an index type:</p>

<p><code>kallisto index -i transcripts.idx transcripts.fasta.gz</code></p>

<h4 id="quantification">Quantification</h4>

<p>Now you can quantify abundances of the transcripts using the two read files reads_1.fastq.gz and reads_2.fastq.gz (the .gz suffix means the read files have been gzipped; <strong>kallisto</strong> can read in either plain-text or gzipped read files). To quantify abundances type:</p>

<p><code>kallisto quant -i transcripts.idx -o output -b 100 reads_1.fastq.gz reads_2.fastq.gz</code></p>

<p>You can also call <strong>kallisto</strong> with</p>

<p><code>kallisto quant -i transcripts.idx -o output -b 100 &lt;(gzcat reads_1.fastq.gz) &lt;(gzcat reads_2.fastq.gz)</code></p>

<p>or with linux, you replace <code>gzcat</code> with <code>zcat</code> or any other program that writes the FASTQ to stdout. This utilizes an additional core to uncompress the FASTQ files, and speeds up the program by 10–15%.</p>

<h4 id="results">Results</h4>

<p>The results of a <strong>kallisto</strong> run are placed in the specified output directory (the -o option), and therefore the test results should be located in the subdirectory “output”. The contents of the directory should look like this:</p>

<pre><code>total 568
-rw-r--r--  1 username  staff  282480 May  3 10:10 abundance.h5
-rw-r--r--  1 username  staff     589 May  3 10:10 abundance.txt
-rw-r--r--  1 username  staff     227 May  3 10:10 run_info.json
</code></pre>

<p>The results of the main quantification, i.e. the abundance estimate using <strong>kallisto</strong> on the data is in the <code>abundance.txt</code> file. Abundances are reported in “estimated counts” (est_counts) and in <a href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/">Transcripts Per Million</a> (TPM). The abundance.txt file you get should look like this:</p>

<pre><code>target_id       length  eff_length      est_counts      tpm
NM_001168316    2283    2105.9  164.133 12856.9
NM_174914       2385    2207.9  1495.6  111741
NR_031764       1853    1675.9  104.27  10263.4
NM_004503       1681    1503.9  332.001 36416.5
NM_006897       1541    1363.9  664     80308.9
NM_014212       2037    1859.9  55      4878.11
NM_014620       2300    2122.9  592.584 46046.7
NM_017409       1959    1781.9  47      4351.04
NM_017410       2396    2218.9  42      3122.41
NM_018953       1612    1434.9  227.995 26210.8
NM_022658       2288    2110.9  4881    381434
NM_153633       1666    1488.9  359.898 39874.2
NM_153693       2072    1894.9  72.5147 6312.74
NM_173860       849     671.903 962     236182
NR_003084       1640    1462.9  0.00787013      0.887453
</code></pre>

<p>The file is tab delimited so that it can easily parsed. It can also be analyzed with the <strong>sleuth</strong> tool.</p>

<p>The <code>run_info.json</code> file contains a summary of the run, including data on the number targets used for quantification, the number of bootstraps performed, the version of the program used and how it was called. You should see this:</p>

<pre><code>{
        "n_targets": 15,
        "n_bootstraps": 100,
        "kallisto_version": "0.42",
        "index_version": 9,
        "start_time": "Sat May  2 15:19:08 2015",
        "call": "kallisto quant -i transcripts.idx -o output -b 100 reads_1.fastq.gz reads_2.fastq.gz"
}
</code></pre>

<p>The h5 file contains the main quantification together with the boostraps in <a href="https://www.hdfgroup.org/HDF5/whatishdf5.html">HDF5 format</a>. The reason for this binary format is to compress the large output of runs with many bootstraps. The <strong>h5dump</strong> command in <strong>kallisto</strong> can be used to convert the file to plain-text.</p>

<p>That’s it.</p>

<p>You can now run kallisto on your dataset of choice. For convenience, we have placed some transcriptome fasta files for human and model organisms <a href="http://bio.math.berkeley.edu/kallisto/transcriptomes/">here</a>. Publicly available RNA-Seq data can be found on the <a href="http://www.ncbi.nlm.nih.gov/sra">short read archive</a> (a convenient mirror and interface to the SRA is available <a href="http://sra.dnanexus.com">here</a>). While <strong>kallisto</strong> cannot process .sra files, such files can be converted to FASTQ with the <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&amp;f=fastq-dump">fastq-dump</a> tool which is part of the <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software">SRA Toolkit</a>.</p>

  </div>
</div>


      </div>
      <hr>
      <footer>
        <p>&copy; 2015 Pachter Lab
          with help from <a href="http://jekyllbootstrap.com" target="_blank" title="The Definitive Jekyll Blogging Framework">Jekyll Bootstrap</a>
          and <a href="http://twitter.github.com/bootstrap/" target="_blank">Twitter Bootstrap</a>
        </p>
      </footer>

    </div>

    


  </body>
</html>

