<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Constructor for a 'sleuth' object</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for sleuth_prep"><tr><td>sleuth_prep</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Constructor for a 'sleuth' object</h2>

<h3>Description</h3>

<p>A sleuth is a group of kallistos. Borrowing this terminology, a 'sleuth' object stores
a group of kallisto results, and can then operate on them while
accounting for covariates, sequencing depth, technical and biological
variance.
</p>


<h3>Usage</h3>

<pre>
sleuth_prep(sample_to_covariates, full_model, filter_fun = basic_filter,
  target_mapping = NULL, max_bootstrap = NULL,
  norm_fun_counts = norm_factors, norm_fun_tpm = norm_factors, ...)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>sample_to_covariates</code></td>
<td>
<p>is a <code>data.frame</code> which contains a mapping
from <code>sample</code> (a column) to some set of experimental conditions or
covariates. The column <code>path</code> is also required, which is a character
vector where each element points to the corresponding kallisto output directory. The column
<code>sample</code> should be in the same order as the corresponding entry in
<code>path</code>.</p>
</td></tr>
<tr valign="top"><td><code>full_model</code></td>
<td>
<p>is a <code>formula</code> which explains the full model (design)
of the experiment OR a design matrix. It must be consistent with the data.frame supplied in
<code>sample_to_covariates</code>.</p>
</td></tr>
<tr valign="top"><td><code>filter_fun</code></td>
<td>
<p>the function to use when filtering.</p>
</td></tr>
<tr valign="top"><td><code>target_mapping</code></td>
<td>
<p>a <code>data.frame</code> that has at least one column
'target_id' and others that denote the mapping for each target. if it is not
<code>NULL</code>, <code>target_mapping</code> is joined with many outputs where it
might be useful. For example, you might have columns 'target_id',
'ensembl_gene' and 'entrez_gene' to denote different transcript to gene
mappings.</p>
</td></tr>
<tr valign="top"><td><code>max_bootstrap</code></td>
<td>
<p>maximum number of bootstrap values to read for each
transcript.</p>
</td></tr>
<tr valign="top"><td><code>norm_fun_counts</code></td>
<td>
<p>a function to perform between sample normalization on the estimated counts.</p>
</td></tr>
<tr valign="top"><td><code>norm_fun_tpm</code></td>
<td>
<p>a function to perform between sample normalization on the TPM</p>
</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
<p>additional arguments passed to the filter function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>sleuth</code> object containing all kallisto samples, metadata,
and summary statistics
</p>


<h3>See Also</h3>

<p><code>sleuth_fit</code> to fit a model, <code>sleuth_test</code> to
test whether a coeffient in the model is zero
</p>


</body></html>
